Applications/Utils/MeshEdit/ExtractSurface.cpp

```diff
@@ -0,0 +1,65 @@
+/**
+ * @brief Extracts the top surface from the given mesh.
+ *
+ * @copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/LICENSE.txt
+ */
+
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "tclap/CmdLine.h"
+
+#include "Applications/ApplicationsLib/LogogSetup.h"
+
+#include "BaseLib/StringTools.h"
+#include "BaseLib/FileTools.h"
+
+#include "FileIO/readMeshFromFile.h"
+#include "FileIO/writeMeshToFile.h"
+
+#include "MathLib/Vector3.h"
+
+#include "MeshLib/Mesh.h"
+#include "MeshLib/MeshSurfaceExtraction.h"
+
+int main (int argc, char* argv[])
+{
+	ApplicationsLib::LogogSetup logog_setup;
+
+	TCLAP::CmdLine cmd("Tool extracts the top surface of the given mesh."
+		, ' ', "0.1");
+	TCLAP::ValueArg<std::string> mesh_in("i", "mesh-input-file",
+		"the name of the file containing the input mesh", true,
+		"", "file name of input mesh");
+	cmd.add(mesh_in);
+	TCLAP::ValueArg<std::string> mesh_out("o", "mesh-output-file",
+		"the name of the file the surface mesh should be written to", false,
+		"", "file name of output mesh");
+	cmd.add(mesh_out);
+
+	cmd.parse(argc, argv);
+
+	std::unique_ptr<MeshLib::Mesh const> mesh(
+	    FileIO::readMeshFromFile(mesh_in.getValue()));
+	INFO("Mesh read: %u nodes, %u elements.", mesh->getNNodes(), mesh->getNElements());
+
+	// extract top surface
+	MathLib::Vector3 const dir(0.0, 0.0, -1.0);
+	double const angle(90);
```

Comment:
I'd add the direction (with default 0,0,-1) and the angle (with default value 90) to the command line arguments. Otherwise the tool should be named 'extractTopSurface'.